### ISSUE TITLE
Help on fusb302 (PD PSU)

### ISSUE THREAD

**Issue by strongtz** [amarula/bsp-rockchip#1]
Can't boot to initramfs using a PD power supply. Problem also occurs in regular mainline kernel.

**Comment by sunielmahesh** [comment]
hi, please elaborate on the specification of this PD power supply.
hope you have flashed initramfs in SPI as per procedure.
Do you have a USB Type-C to USB-A 2.0 Male Cable ? if yes please try powering it up with that. 
Thanks

**Comment by strongtz** [comment]
I'm using a 30W PD PSU (5V@3A 9V@2A 15V@2A 20V@1.5A)
In mainline kernel, once I append "modprobe.blacklist=fusb302", it boots up just fine, so I think the driver has issues.
Non-PD PSU also works.
Problem also described in Armbian forum.
[https://forum.armbian.com/topic/7552-roc-rk3399-pc-renegade-elite/](url)

**Comment by sunielmahesh** [comment]
Hi, I have a 18W PD PSU(5V@3A 9V@2A 12V@1.5A ) USB TYPE A adapter. I am using a USB Type-C to USB-A 2.0 Male Cable with the adapter.
I am booting initramfs from SPI fine with out any issues.
The images at https://github.com/amarula/bsp-rockchip/ are working images.
Are you building your own initramfs images ?
If so please share the version of the sources (ATF, uboot, linux) and also build framework used. Thanks

**Comment by strongtz** [comment]
Are you sure your PSU is a PD one? As far as I know, your PSU is more likely a QC charger. Therefore, PD will not be activated and there's no issue. I'm using uboot and initramfs from this repo.

**Comment by strongtz** [comment]
BTW, PD only works with a C-to-C cable.

**Comment by sunielmahesh** [comment]
yes, mine is a18W USB TYPE A adapter. Its not a PD. sorry for that.
I will try to find a PD and test if I get hold of any. will update. Thanks

**Comment by sunielmahesh** [comment]
Hi strongtz, i don't have a PD with me as of now. Its a lockdown at my place here. I will update once 
i get a PD device and post the results. Thanks

**Comment by karendar** [comment]
This was discussed in detail on the Armbian forums. Here is a discussion on the issue: 

https://lore.kernel.org/linux-usb/0ac6bbe7-6395-526d-213c-ac58a19d8673@fivetechno.de/
https://lore.kernel.org/linux-usb/9d4b25e4-0b88-bf3d-7265-e43026325e2d@fivetechno.de/
